addons/rmsmartshape/shapes/shape.gd

```diff
@@ -1670,50 +1691,55 @@ func _build_edge_with_material(
-					var offset: Vector2 = delta_normal * taper_size
-
-					taper_quad.pt_d = taper_quad.pt_a + offset
-					taper_quad.pt_c = taper_quad.pt_b + offset
-					new_quad.pt_a = taper_quad.pt_d
-					new_quad.pt_b = taper_quad.pt_c
-					new_quads.push_front(taper_quad)
-				# If a new taper quad doesn't fit, re-texture the new_quad
-				else:
-					new_quad.texture = taper_texture
+			did_taper_left = true
+			var taper_quad := _taper_quad_left(new_quad, edge_material, texture_idx)
+			if taper_quad != null:
+				new_quads.push_front(taper_quad)
 		if is_last_tess_point and edge_material != null and edge_material.use_taper_texture:
-			var taper_texture: Texture2D = edge_material.get_texture_taper_right(texture_idx)
-			if taper_texture != null:
-				var taper_size: Vector2 = taper_texture.get_size()
-				var fit: bool = absf(taper_size.x) <= new_quad.get_length_average()
-				if fit:
-					var taper_quad := new_quad.duplicate()
-					taper_quad.corner = 0
-					taper_quad.texture = taper_texture
-					var delta_normal: Vector2 = (taper_quad.pt_d - taper_quad.pt_a).normalized()
-					var offset: Vector2 = delta_normal * taper_size
-					taper_quad.pt_a = taper_quad.pt_d - offset
-					taper_quad.pt_b = taper_quad.pt_c - offset
-					new_quad.pt_d = taper_quad.pt_a
-					new_quad.pt_c = taper_quad.pt_b
-					new_quads.push_back(taper_quad)
-				# If a new taper quad doesn't fit, re-texture the new_quad
+			did_taper_right = true
+			var taper_quad := _taper_quad_right(new_quad, edge_material, texture_idx)
+			if taper_quad != null:
+				new_quads.push_back(taper_quad)
+
+		# Taper sharp corners
+		#is_not_corner = new_quads[0].corner == SS2D_Quad.CORNER.NONE
+		if taper_sharp:
+			if sharp_taper_next != null && is_not_corner:
+				var taper := _taper_quad_right(sharp_taper_next, edge_material, texture_idx)
+				if taper != null:
+					taper.ignore_weld_next = true
+					edge.quads.push_back(taper)
 				else:
-					new_quad.texture = taper_texture
+					sharp_taper_next.ignore_weld_next = true
+			sharp_taper_next = null
+			var vert := verts[vert_idx]
+			var prev_vert := verts[wrapi(vert_idx - 1, 0, verts.size() - 1)] as Vector2
+			var next_vert := verts[wrapi(vert_idx + 1, 0, verts.size() - 1)] as Vector2
+			if !did_taper_left && is_not_corner:
+				var ang_from := prev_vert.angle_to_point(vert)
+				var ang_to = vert.angle_to_point(next_vert)
+				var ang_dif = angle_difference(ang_from, ang_to)
+				if absf(ang_dif) > PI - PI * 0.5:
+					var taper = _taper_quad_left(new_quad, edge_material, texture_idx)
+					if taper != null:
+						new_quads.push_front(taper)
+			if !did_taper_right:
+				var next_next_vert := verts[wrapi(vert_idx + 2, 0, verts.size() - 1)] as Vector2
+				var ang_from = vert.angle_to_point(next_vert)
+				var ang_to = next_vert.angle_to_point(next_next_vert)
+				var ang_dif = angle_difference(ang_from, ang_to)
+				if absf(ang_dif) > PI - PI * 0.5:
+					sharp_taper_next = new_quad
```

Comment:
- Various missing types. Use `:=` please.
- `as Vector2` not needed
- `!`, `&&`, `||`
- `PI - PI * 0.5` -> `PI * 0.5`